executor/src/witgen/global_constraints.rs

```diff
@@ -280,16 +275,33 @@ fn propagate_constraints<T: FieldElement>(
                 false
             }
         }
-        Identity::Lookup(LookupIdentity { left, right, .. })
-        | Identity::PhantomLookup(PhantomLookupIdentity { left, right, .. }) => {
-            if !left.selector.is_one() || !right.selector.is_one() {
+        Identity::BusReceive(..) => false,
+        Identity::BusSend(bus_interaction) => {
+            let send = bus_interaction;
+            let receive = match bus_interaction.try_match_static(bus_receives) {
+                Some(receive) => receive,
+                None => {
+                    return false;
+                }
+            };
```

Comment:
```suggestion
            let receive = bus_interaction.try_match_static(bus_receives).ok_or(false)?;
```

Reply:
Btw, this could be precomputed and stored in `BusSend`, in which case we do not need to pass `bus_receives` to this function

Reply:
Yes, I want to keep the dynamic case in mind already. I mean I can also have an `static_match: Option<&BusReceive>` or something, but let's see.